### DISCUSSION TITLE
Minting with Solana

### DISCUSSION THREAD

**Discussion by idiglove** [cnp0/repomint#25] [General]
I will put out steps, tutorials and resources I've found in the web to carry out this project

### Wallet
There are ways to create a SOL wallet, I chose Phantom: https://phantom.app/

### Resources
1. Full Stack with Solana - https://dev.to/dabit3/the-complete-guide-to-full-stack-solana-development-with-react-anchor-rust-and-phantom-3291
2. Minting with Candy Machine (mostly through CLI)  - https://www.youtube.com/watch?v=TjZmq1d8Nyg 
3. Mint with Solana - https://www.quicknode.com/guides/web3-sdks/how-to-mint-an-nft-on-solana 
   - I'm actually leaning more towards doing this, as it goes through a backend and it's JS
4. Mint with Solana YT tutorial -  https://www.youtube.com/watch?v=L4WWQzOBNIg

### Issues / Blockers
- How to actually pass the id of the generative art to this createMint function:
```
  let mint = await splToken.Token.createMint(
    connection,
    fromWallet,
    fromWallet.publicKey,
    null,
    9,
    splToken.TOKEN_PROGRAM_ID,
  );
  ```

## From doing Resources #3 
I tinkered it a little bit to use my wallets, and I was able to mint and send the toke
![Screen Shot 2021-09-18 at 4 04 09 PM](https://user-images.githubusercontent.com/29911508/133881748-a403c9a2-e821-4235-9de6-672f580e9fd3.png)
Now what's left is actually linking it to a generative art

**Comment by cnpryer** [comment]
Awesome! Discussions. Didn't even realize you enabled this. I love this. I was reading about the quiknode info you shared. Still defer to you. Whatever you think makes the most sense. 

I do have a question tho. Do we want to capture this in our issues or should we keep this here? Idk how I missed this.

**Comment by cnpryer** [comment]
Not sure how this translates to javascript, but from [docs.metaplex.com](https://docs.metaplex.com/#overview-1)

```rust

#[repr(C)]
#[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug, Clone)]
pub struct Data {
    /// The name of the asset
    pub name: String,
    /// The symbol for the asset
    pub symbol: String,
    /// URI pointing to JSON representing the asset
    pub uri: String,
    /// Royalty basis points that goes to creators in secondary sales (0-10000)
    pub seller_fee_basis_points: u16,
    /// Array of creators, optional
    pub creators: Option<Vec<Creator>>,
}
```

It looks like in the metadata of the protocol we use we'd just add a link to the digital asset. So maybe instead of paying for hosting for now we can prob get away with using discord cdn. But I'm not sure.

**Reply by Pfed-prog** [reply]
Pinata.cloud has a free plan for 1gb of data

**Comment by Pfed-prog** [comment]
https://solible.com/#/ (Solana NFT gallery) seems to be built with Serum  https://github.com/project-serum/awesome-serum. It and metaplex developer guide seems to lead to https://spl.solana.com/token#example-create-a-non-fungible-token
